plugins/package-managers/node/src/main/kotlin/pnpm/Pnpm.kt

```diff
@@ -170,3 +172,21 @@ private fun ModuleInfo.getScopeDependencies(scope: Scope) =
 
         Scope.DEV_DEPENDENCIES -> devDependencies.values.toList()
     }
+
+/**
+ * Find the [List] of [ModuleInfo] objects for the project in the given [workingDir]. If there are nested projects,
+ * the `pnpm list` command yields multiple arrays with modules. In this case, only the top-level project should be
+ * analyzed. This function tries to detect the corresponding [ModuleInfo]s based on the [workingDir]. If this is not
+ * possible, as a fallback the first list of [ModuleInfo] objects is returned.
+ */
+private fun Sequence<List<ModuleInfo>>.findModules(workingDir: File): List<ModuleInfo> {
+    val moduleInfoIterator = iterator()
+    val first = moduleInfoIterator.nextOrNull() ?: return emptyList()
+
+    fun List<ModuleInfo>.matchesWorkingDir() = any { File(it.path).absoluteFile == workingDir }
```

Comment:
Would the following do the same?
```
private fun Sequence<List<ModuleInfo>>.findModules(workingDir: File): List<ModuleInfo> = 
  first { list ->
      list.any { File(it.path).absoluteFile == workingDir }
  }
```

Reply:
I don't think so. This will not handle the corner case that no element in the sequence satisfies the condition. The current code then returns the first element as fallback, while the proposed one will fail, right?

Reply:
Hm, and this one?
```
private fun Sequence<List<ModuleInfo>>.findModules(workingDir: File): List<ModuleInfo> = 
  firstOrNull { list ->
      list.any { File(it.path).absoluteFile == workingDir }
  }.orEmpty()
```

Reply:
> I would actually prefer to make the assumption that there is a match, but would also like to hear @sschuberth if he agrees too.

I currently don't have the time to think about this filter logic in detail.